lib/classes/Variables.js

```diff
@@ -155,28 +155,57 @@ class Variables {
       .replace(this.fileRefSyntax, (match, varName) => varName.trim());
     const referencedFileFullPath = path.join(this.serverless.config.servicePath,
       referencedFileRelativePath);
+    let fileExtension = referencedFileRelativePath.split('.');
+    fileExtension = fileExtension[fileExtension.length - 1];
 
+    // Validate file exists
     if (!this.serverless.utils.fileExistsSync(referencedFileFullPath)) {
       return undefined;
     }
 
-    let valueToPopulate = this.serverless.utils.readFileSync(referencedFileFullPath);
-    if (matchedFileRefString !== variableString) {
-      let deepProperties = variableString
-        .replace(matchedFileRefString, '');
-      if (deepProperties.substring(0, 1) !== ':') {
+    let valueToPopulate;
+
+    // Process JS files
+    if (fileExtension === 'js') {
+      const jsFile = require(referencedFileFullPath); // eslint-disable-line global-require
+      let jsModule = variableString.split(':')[1];
+      jsModule = jsModule.split('.')[0];
+      valueToPopulate = jsFile[jsModule]();
+      let deepProperties = variableString.replace(matchedFileRefString, '');
+      deepProperties = deepProperties.slice(1).split('.');
+      deepProperties.splice(0, 1);
+      valueToPopulate = this.getDeepValue(deepProperties, valueToPopulate);
+
+      if (typeof valueToPopulate === 'undefined') {
         const errorMessage = [
           'Invalid variable syntax when referencing',
-          ` file "${referencedFileRelativePath}" sub properties`,
-          ' Please use ":" or "." to reference sub properties.',
+          ` file "${referencedFileRelativePath}".`,
+          ' Check your javascript is returning the correct data.',
         ].join('');
         throw new this.serverless.classes
           .Error(errorMessage);
       }
-      deepProperties = deepProperties.slice(1).split('.');
-      valueToPopulate = this.getDeepValue(deepProperties, valueToPopulate);
     }
 
+    // Process everything except JS
+    if (fileExtension !== 'js') {
+      valueToPopulate = this.serverless.utils.readFileSync(referencedFileFullPath);
+      if (matchedFileRefString !== variableString) {
+        let deepProperties = variableString
+          .replace(matchedFileRefString, '');
+        if (deepProperties.substring(0, 1) !== ':') {
+          const errorMessage = [
+            'Invalid variable syntax when referencing',
+            ` file "${referencedFileRelativePath}" sub properties`,
+            ' Please use ":" or "." to reference sub properties.',
```

Comment:
Is this error message accurate?  Because at the top there's a an `if` which only checks for `:`.